dali/operators/reader/loader/file_label_loader.h

```diff
@@ -47,21 +47,47 @@ class DLL_PUBLIC FileLabelLoader : public Loader<CPUBackend, ImageLabelWrapper>
  public:
   explicit inline FileLabelLoader(
     const OpSpec& spec,
-    vector<std::pair<string, int>> image_label_pairs = std::vector<std::pair<string, int>>(),
     bool shuffle_after_epoch = false)
     : Loader<CPUBackend, ImageLabelWrapper>(spec),
-      file_root_(spec.GetArgument<string>("file_root")),
-      file_list_(),
-      image_label_pairs_(std::move(image_label_pairs)),
       shuffle_after_epoch_(shuffle_after_epoch),
       current_index_(0),
       current_epoch_(0) {
 
+      if (!spec.TryGetArgument(file_root_, "file_root") && !spec.HasArgument("files")) {
+        DALI_FAIL("``file_root`` argument is required when not supplygin explicit file paths "
+          "through ``files`` argument");
+      }
+
+      DALI_ENFORCE(spec.HasArgument("files") + spec.HasArgument("file_list") <= 1,
+        "File paths can be provided through ``files`` or ``file_list`` but not both.");
+
+      DALI_ENFORCE(spec.HasArgument("files") || !spec.HasArgument("labels"),
+        "The argument ``labels`` is valid only when file paths "
+        "are provided as ``files`` argument.");
```

Comment:
As mentioned before, that should probably be part of the doc as well.

Reply:
> Labels accompanying contents of files listed in ``files`` argument.

I think that sort-of indicates that _having_ files argument is a prerequisite. And I'm against writing a documentation for robots.

Reply:
I mean more about files vs file_list.